src/map/VisualStimuli.kt

```diff
@@ -0,0 +1,11 @@
+package map
+
+class VisualStimuli(nodesLookingUp: List<Node>,
+                    nodesLookingDown: List<Node>,
+                    nodesLookingLeft: List<Node>,
+                    nodesLookingRight: List<Node>) {
+    val nodesLookingUp: List<Node> = nodesLookingUp
```

Comment:
kotlin has data classes for structures like these.
also, this is pretty rigid, what if you want to include all nodes within certain radius (which is actually more natural), not just ones on the same x- or y- axes.